File: TesteTorniquete/src/Torniquete.java
public class Torniquete {
    private Detector detector;
    private Trinco trinco;
    private Sirene sirene;
    private MaquinaEstados maquinaEstados;

    public Torniquete(){
        this.detector = new Detector();
        this.trinco = new Trinco();
        this.sirene = new Sirene();
        maquinaEstados = new MaquinaEstados();
    }

    public void executar(){
        EventoTorniquete evento = null;
        do {
            evento = detector.detectar();
            Accao accao = maquinaEstados.processar(evento);
            System.out.println("ACCAO: " + accao);
            switch(accao){
                case bloquear: trinco.bloquear(); break;
                case desbloquear: trinco.desbloquear(); break;
                case activar_sirene: sirene.activar(); break;
                case desactivar_sirene: sirene.desactivar(); break;

            }
        }
        while(evento != EventoTorniquete.REINICIAR);
    }
}

File: TesteTorniquetePython/torniquete/torniquete.py
from torniquete.maquinaestados import MaquinaEstados
from torniquete.sirene import Sirene
from torniquete.detector import Detector
from torniquete.trinco import Trinco
from torniquete.evento import Evento
from torniquete.accao import Accao


class Torniquete():
    def __init__(self):
        self._maqEst = MaquinaEstados()
        self._detector = Detector()
        self._trinco = Trinco()
        self._sirene = Sirene()

    def _processar(self, evento):
        pass

    def executar(self):
        evento = None
        while evento != Evento.REINICIAR:
            evento = self._detector.detectar()
            accao = self._maqEst.processar(evento)
            if accao == Accao.bloquear:
                self._trinco.bloquear()
            elif accao == Accao.desbloquear:
                self._trinco.desbloquear()
            elif accao == Accao.activar_sirene:
                self._sirene.activar()
            elif accao == Accao.desactivar_sirene:
                self._sirene.desactivar()
